### PR TITLE
Transfer learning

### PR DESCRIPTION
**Pull request by SinaSarparast**
- Removed one from the classifier layer at the end of the Resnet18
- Added scheduler to ResNet18 model
- Added the momentum hyper-parameter.
- merged with the main branch

### CODE REVIEW

File: image_classification_simulation/models/resnet_baseline.py [hunk 1]
```diff
@@ -33,19 +35,31 @@ def __init__(self, hyper_params: typing.Dict[typing.AnyStr, typing.Any]):
...
+        """Configures the optimizer.
+
+        Returns
+        -------
+        typing.Any
+            The optimizer.
+        """
+        optimizer = load_optimizer(self.hparams, self)
+        scheduler = lr_scheduler.StepLR(
+            optimizer, step_size=0.9, gamma=0.1
```
Comment (SinaSarparast) [inline image_classification_simulation/models/resnet_baseline.py]: The hyper-parameters should be specified in the config file. I avoided them for now since adding then requires adding hyper-parameter checking here. We need to have a centralized hyper-parameter validation that sets default values for parameters not specified in the config file.
